### PR TITLE
Fix missing standard includes, reduce warnings in doxygen-enabled builds

### PR DESCRIPTION
**Pull request by aPonza**
> I can fix the namespace 'std' stuff tomorrow

from https://github.com/PointCloudLibrary/pcl/pull/3744#issuecomment-599153621

This reduces the doxygen-clang log by 1075 lines, bringing it to 16880 if we consider the ones from #3701 as done. There is absolutely no dependency between these two PRs though.

### CODE REVIEW

File: common/include/pcl/common/impl/bivariate_polynomial.hpp [hunk 1]
```diff
@@ -40,6 +40,12 @@
 #define BIVARIATE_POLYNOMIAL_HPP
 
 #include <algorithm>
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <istream>
+#include <ostream>
```
Comment (kunaltyagi) [inline common/include/pcl/common/impl/bivariate_polynomial.hpp]:
```suggestion
#include <iostream>
```

File: filters/include/pcl/filters/impl/convolution_3d.hpp [hunk 2]
```diff
@@ -37,6 +37,11 @@
  *
  */
 
+#include <cmath>
```
Comment (kunaltyagi) [inline filters/include/pcl/filters/impl/convolution_3d.hpp]: Move these after the pcl includes down on L50

File: recognition/include/pcl/recognition/impl/ransac_based/simple_octree.hpp [hunk 3]
```diff
@@ -8,392 +8,365 @@
...
-: data_ (nullptr),
-  parent_ (nullptr),
-  children_(nullptr)
-{}
-
-//===============================================================================================================================
-
-template<typename NodeData, typename NodeDataCreator, typename Scalar> inline
-pcl::recognition::SimpleOctree<NodeData, NodeDataCreator, Scalar>::Node::~Node ()
+namespace pcl
```
Comment (kunaltyagi) [inline recognition/include/pcl/recognition/impl/ransac_based/simple_octree.hpp]: Other files don't have this change. Maybe restrict namespace addition to the existing PR.

File: common/include/pcl/common/impl/bivariate_polynomial.hpp (hunk 1 above)
Comment (aPonza) [reply common/include/pcl/common/impl/bivariate_polynomial.hpp]:
```suggestion
#include <fstream>
#include <iostream>
```
[fstream](https://en.cppreference.com/w/cpp/header/fstream) doesn't include [iostream](https://en.cppreference.com/w/cpp/header/iostream) if that's what you mean.

File: recognition/include/pcl/recognition/impl/ransac_based/simple_octree.hpp (hunk 3 above)
Comment (aPonza) [reply recognition/include/pcl/recognition/impl/ransac_based/simple_octree.hpp]: This came out of a conflict whilst cherry-picking. I couldn't understand what was happening, now I do, thanks!

**Comment by aPonza** [comment]
```
d:\a\1\s\surface\include\pcl\surface\impl\surfel_smoothing.hpp(84): fatal error C1083: Cannot open compiler intermediate file: 'pcl_surface.dir\Release\surfel_smoothing.obj': Not enough space [D:\a\build\surface\pcl_surface.vcxproj]
LINK : fatal error LNK1257: code generation failed [D:\a\build\surface\pcl_surface.vcxproj]
```

[link](https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=13144&view=logs&j=ac8ecbfd-de88-5804-0904-342030d342ff&t=6e1dc337-a29b-51eb-250a-74a1941e72c5&l=982)

File: recognition/include/pcl/recognition/surface_normal_modality.h [hunk 4]
```diff
@@ -45,6 +45,14 @@
 #include <pcl/point_types.h>
 #include <pcl/features/linear_least_squares_normal.h>
 
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+// #include <iostream>
```
Comment (SergioRAgostinho) [inline recognition/include/pcl/recognition/surface_normal_modality.h]: Any reason to keep the comment around?

Comment (aPonza) [reply recognition/include/pcl/recognition/surface_normal_modality.h]: Commented lines like https://github.com/PointCloudLibrary/pcl/blob/9427da1fac4088e61e037cd3e0d4dedbdfebcd0b/recognition/include/pcl/recognition/surface_normal_modality.h#L1122

**Pull request merged**